frontend/src/components/DrawComponent.jsx

```diff
@@ -0,0 +1,58 @@
+import React, { useEffect, useRef, useState } from 'react';
+import Draw from 'ol/interaction/Draw.js';
+import { unByKey } from 'ol/Observable.js';
+import VectorLayer from "ol/layer/Vector";
+import VectorSource from "ol/source/Vector";
+
+const DrawComponent = ({ map, geometryType, onDeleteGeometries }) => {
+  const drawInteraction = useRef(null);
+  const [drawing, setDrawing] = useState(false);
+
+  useEffect(() => {
+    if (!map || !geometryType) return;
+
+    const drawLayer = new VectorLayer({
+      name: 'drawLayer',
```

Comment:
While I included this code myself, it's advisable to avoid using literals like 'drawLayer' in your code. Please do export them into a seperate file (or files...)